oligo_designer_toolsuite/IO/_database.py

```diff
@@ -325,23 +311,33 @@ def create_target_region(region, genes):
                 )
             return file_region_annotation
 
-        Path(dir_output).mkdir(parents=True, exist_ok=True)
-        self.file_oligos_DB = os.path.join(
-            dir_output,
-            f"oligo_DB_{self.species}_{self.genome_assembly}_{self.annotation_source}_release_{self.annotation_release}_{region}",
-        )
+        self.file_name_oligos_DB_tsv = f"oligo_DB_{self.species}_{self.genome_assembly}_{self.annotation_source}_release_{self.annotation_release}_{region}.tsv"
+        self.file_name_oligos_DB_gtf = f"oligo_DB_{self.species}_{self.genome_assembly}_{self.annotation_source}_release_{self.annotation_release}_{region}.gtf"
+        self.file_name_oligos_DB_fasta = f"oligo_DB_{self.species}_{self.genome_assembly}_{self.annotation_source}_release_{self.annotation_release}_{region}.fasta"
         file_region_annotation = create_target_region(region, genes)
         if genes is None:
             genes = self.gene_transcript.get_genes_from_annotation()
         self.oligos_DB = self.oligos.generate(
-            file_region_annotation, genes, number_batchs, dir_output
+            file_region_annotation, genes, n_jobs, self.dir_annotation
         )
-        if write:
-            self.write_oligos_DB()
-
         # clean folder
         os.remove(file_region_annotation)
 
+    def remove_genes_with_insufficient_probes(self, pipeline_step, write=True):
+        """Deletes from the `oligo_DB` the genes which do not have any more probes, and writes them in a file with the name of the step of teh pipeline at which they have been deleted.
```

Comment:
...and **optionally** writes them in a file with the name of the step of **the** pipeline at...